### PR TITLE
feat: [18] - add update Course status command & contract & policy

### PR DESCRIPTION
**Pull request by tomczak-mateusz**
- update Rubocop max line length to 180
- add UpdateStatus course command & spec
- add UpdateStatus course contract & spec
- add update_status? policy to Course Policy & spec

### CODE REVIEW

File: app/policies/course_policy.rb [hunk 1]
```diff
@@ -23,6 +23,11 @@ def destroy?
     record.draft? && is_teacher_and_course_author?(record)
   end
 
+  def update_status?
```
Comment (Thevel0per) [inline app/policies/course_policy.rb:26]: Here we should just check if user is either an admin or an author of the course. We should not check status as it is part of the validation not authorization.

File: spec/commands/courses/update_status_spec.rb [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          before { course.update!(author_id: current_user.id) }
+
+          it 'does not raise an error' do
+            expect { result }.not_to raise_error
+          end
+
+          it 'does updates provided course with provided status' do
+            result rescue nil
+            course.reload
+            expect(course.status).to eq('ready')
```
Comment (Thevel0per) [inline spec/commands/courses/update_status_spec.rb:83]:
```suggestion
            expect { result }.to change { course.status }.from('draft').to('ready')
```

Then you can replace all three lines with just this one. Also doing `result rescue nil` here doesn't do anything as you wont get any error thrown.

**Review by Thevel0per — CHANGES_REQUESTED** [review]

File: app/policies/course_policy.rb (hunk 1 above)
Comment (tomczak-mateusz) [reply app/policies/course_policy.rb:26]: fixed for both update endpoints

File: spec/commands/courses/update_status_spec.rb (hunk 2 above)
Comment (tomczak-mateusz) [reply spec/commands/courses/update_status_spec.rb]: fixed

**Review by Thevel0per — APPROVED** [review]

**Pull request merged**